dcap-ql/src/quote.rs

```diff
@@ -249,6 +308,43 @@ impl<'a> Qe3CertData<'a> for Qe3CertDataPpid<'a> {
     }
 }
 
+impl<'a> Qe3CertData<'a> for RawQe3CertData<'a>{
+    fn parse(_type_: CertificationDataType, data: &Cow<'a, [u8]>) -> Result<Self> {
+        Ok(data.clone())
+    }
+}
+
+impl<'a> Qe3CertData<'a> for Qe3CertDataPckCertChain<'a> {
+    fn parse(type_: CertificationDataType, data: &Cow<'a, [u8]>) -> Result<Self>{
+        if type_ != CertificationDataType::PckCertificateChain {
+            bail!("Invalid certification data type: {:?}", type_);
+        }
+
+        let mut data = match data {
+             Cow::Borrowed(s) =>
+                 std::str::from_utf8(s)
+                     .map(Cow::Borrowed)
+                     .map_err(|e| format_err!("Invalid certificate format: {}", e))?,
+             Cow::Owned(s) =>
+                 String::from_utf8(s.to_vec())
+                     .map(Cow::Owned)
+                     .map_err(|e| format_err!("Invalid certificate format: {}", e))?,
+        };
+		// TODO: use pkix PemBlock parser
+        let mut certs = vec![];
+        let mark = "-----END CERTIFICATE-----";
+        while let Some(pos) = data.find(mark) {
+            certs.push(data.take_prefix(pos + mark.len()).expect("validated -- pos is always valid"));
+            if let Some(start) = data.find("-") {
+                data.take_prefix(start).expect("validated -- start is always valid");
```

Comment:
I think `expect` is actually more confusing here than `unwrap` (or at least I found the expect text confusing)